canisters/src/vts/src/lib.rs

```diff
@@ -116,15 +146,21 @@ fn get_user() -> VTSResult<User> {
 #[ic_cdk::update]
 fn request_firmware() -> VTSResult<()> {
     let caller = ic_cdk::api::caller();
-    ic_cdk::println!("{} is requested firmware", caller);
-    FIRMWARE_REQUESTS.with(|requests| {
-        if requests.borrow_mut().contains_key(&caller) {
-            return Err(Error::AlreadyExists);
+    ADMINS.with(|admins| {
```

Comment:
We need to provider authentication to all our canister methods. So only our users or customers can use this canister. We need to implement methods to add and delete users (admin actions). So in almost every canister method you need to check that caller exists in USERS map. So before users can use our system we need to register them by adding them to USERS map by admin.